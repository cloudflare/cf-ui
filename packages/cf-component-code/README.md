# cf-component-code

> Cloudflare Code Component

## Installation

```sh
$ npm install cf-component-code
```

## Usage

```js
const React = require('react');
const {Code, CodeBlock} = require('cf-component-code');

export default class Application extends React.Component {
  render() {
    return (
      <div>
        <Code>Hello World</Code>
        <CodeBlock>Hello World</CodeBlock>
      </div>
    );
  }
}
```
