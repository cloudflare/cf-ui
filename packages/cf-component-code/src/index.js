import { applyTheme } from 'cf-style-container';

import CodeUnstyled from './Code';
import CodeTheme from './CodeTheme';
import CodeBlockUnstyled from './CodeBlock';
import CodeBlockTheme from './CodeBlockTheme';

const Code = applyTheme(CodeUnstyled, CodeTheme);
const CodeBlock = applyTheme(CodeBlockUnstyled, CodeBlockTheme);

export {
  Code,
  CodeUnstyled,
  CodeTheme,
  CodeBlock,
  CodeBlockUnstyled,
  CodeBlockTheme
};
