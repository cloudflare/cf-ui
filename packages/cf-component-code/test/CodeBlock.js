import React from 'react';
import renderer from 'react-test-renderer';
import { CodeBlock } from 'cf-component-code';

test('should render', () => {
  const component = renderer.create(<CodeBlock>Hello</CodeBlock>);
  expect(component.toJSON()).toMatchSnapshot();
});
