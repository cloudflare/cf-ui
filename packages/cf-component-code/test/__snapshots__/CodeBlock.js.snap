// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<pre
  className="cf-code__block"
>
  <code
    className="cf-code"
  >
    Hello
  </code>
</pre>
`;
